numpy/core/src/multiarray/item_selection.c

```diff
@@ -21,11 +21,171 @@
 #include "lowlevel_strided_loops.h"
 #include "array_assign.h"
 
-#include "item_selection.h"
 #include "npy_sort.h"
 #include "npy_partition.h"
 #include "npy_binsearch.h"
 #include "alloc.h"
+#include "arraytypes.h"
+
+
+
+static NPY_INLINE int
+npy_fasttake_impl(
```

Comment:
Elsewhere I've seen we have an `-O3` annotation for gcc. If we don't compile at that level by default, can we use that annotation here?

Reply:
I had to put the attribute on the `npy_fasttake` function and not on the `_impl` one (or maybe the NPY_INLINE has to have a different order (come first). But I think it broke the inlinening first. In any case that works now. The slowdown with complex256 is still there (no idea...), but the general PutMask slowdown is much less/gone and some things got quite a bit faster.

Reply:
(the timings in the above comment are updated.)